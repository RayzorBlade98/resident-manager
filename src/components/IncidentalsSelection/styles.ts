import { SxProps, Theme } from '@mui/material';
import { defaultColors } from '../../types/styles';

const styles: { [id: string]: SxProps<Theme> } = {
  list: { width: '100%', bgcolor: defaultColors.mainLight1 },
};

export default styles;
