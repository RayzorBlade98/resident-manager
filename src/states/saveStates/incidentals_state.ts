import { atom } from 'recoil';
import { setRecoil } from 'recoil-nexus';
import { Incidentals } from 'src/types/incidentals';

/**
 * List of all incidentals
 */
export type IncidentalsState = Incidentals[];

/**
 * Incidentals recoil state
 */
export const incidentalsState = atom<IncidentalsState>({
  key: 'incidentalsState',
  default: [],
});

export abstract class IncidentalsStateManager {
  /**
   * Adds new incidentals to the incidentals state
   * @param incidentals new incidentals that should be added
   */
  public static addIncidentals(incidentals: Incidentals): void {
    setRecoil(incidentalsState, (state: IncidentalsState) => [
      ...state,
      incidentals,
    ]);
  }
}

export default IncidentalsState;
