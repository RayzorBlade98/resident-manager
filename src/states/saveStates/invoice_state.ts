import { atom } from 'recoil';
import { setRecoil } from 'recoil-nexus';
import { Invoice } from '_/types/invoice';

/**
 * The invoice state is a list of all invoices
 */
export type InvoiceState = Invoice[];

/**
 * Invoice state
 */
export const invoiceState = atom<InvoiceState>({
  key: 'invoiceState',
  default: [],
});

/**
 * Class that provides utility functions to manage the invoice state
 */
export abstract class InvoiceStateManager {
  /**
   * Adds a new invoice to the invoice state
   * @param invoice new invoice that should be added
   */
  public static addInvoice(invoice: Invoice): void {
    setRecoil(invoiceState, (state: InvoiceState) => [...state, invoice]);
  }
}
