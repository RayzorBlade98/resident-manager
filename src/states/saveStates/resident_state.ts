import { selector } from 'recoil';
import { setRecoil } from 'recoil-nexus';
import { v4 as uuid } from 'uuid';
import saveState, { SaveState } from './save_state';
import { Resident } from '_/types/resident';
import { getCurrentMonthYear } from '_/types/date';

/**
 * The resident state is a list of all residents
 */
export type ResidentState = Resident[];

export function defaultResidentsState(): ResidentState {
  // return [];
  const dummyResidents: ResidentState = [];
  for (let i = 0; i < 8; i += 1) {
    dummyResidents.push({
      id: uuid(),
      firstName: 'Max',
      lastName: 'Mustermann',
      rent: 50000,
      invoiceStart: getCurrentMonthYear(),
    });
  }
  return dummyResidents;
}

/**
 * Selector for the resident state
 */
export const residentState = selector<ResidentState>({
  key: 'residentState',
  get: ({ get }) => {
    const state = get(saveState);
    return state.residents;
  },
});

/**
 * Adds a new resident to the resident state
 * @param resident new resident that should be added
 */
export function addResident(resident: Resident): void {
  setRecoil(saveState, (state: SaveState) => ({
    ...state,
    residents: [...state.residents, resident],
  }));
}

export default residentState;
