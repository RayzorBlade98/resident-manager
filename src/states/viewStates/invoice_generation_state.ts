import { atom, selector } from 'recoil';
import { setRecoil } from 'recoil-nexus';
import { Incidentals } from '_/types/incidentals';

/**
 * Invoice generation state
 */
export interface InvoiceGenerationState {
  /**
   * List of all selected incidentals
   */
  selectedIncidentals: Incidentals[];
}

/**
 * Invoice generation recoil state
 */
export const invoiceGenerationState = atom<InvoiceGenerationState>({
  key: 'invoiceGenerationState',
  default: {
    selectedIncidentals: [],
  },
});

/**
 * Selector for the selected incidentals
 */
export const selectedInvoiceIncidentalsState = selector<Incidentals[]>({
  key: 'invoiceGenerationState-selectedIncidentals',
  get: ({ get }) => get(invoiceGenerationState).selectedIncidentals,
});

/**
 * Adds new incidentals to the list of selected incidentals
 * @param incidentals incidentals that should be added
 */
export function addSelectedIncidentals(incidentals: Incidentals): void {
  setRecoil(invoiceGenerationState, (state) => ({
    ...state,
    selectedIncidentals: [...state.selectedIncidentals, incidentals],
  }));
}

/**
 * Removes incidentals from the list of selected incidentals
 * @param incidentals incidentals that should be removed
 */
export function removeSelectedIncidentals(incidentals: Incidentals): void {
  setRecoil(invoiceGenerationState, (state) => ({
    ...state,
    selectedIncidentals: state.selectedIncidentals.filter(
      (i) => i.id !== incidentals.id,
    ),
  }));
}

export default invoiceGenerationState;
