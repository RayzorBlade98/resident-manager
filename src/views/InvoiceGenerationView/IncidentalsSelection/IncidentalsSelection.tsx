import {
  Checkbox,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
} from '@mui/material';
import React from 'react';
import { useRecoilValue } from 'recoil';
import { incidentalsState } from '_/states/saveStates/incidentals_state';
import { Incidentals } from '_/types/incidentals';
import {
  addSelectedIncidentals,
  removeSelectedIncidentals,
  selectedInvoiceIncidentalsState,
} from '_/views/InvoiceGenerationView/states/invoice_generation_view_state';

/**
 * Component so select incidentals that should be included to the invoice
 */
function IncidentalsSelection(): JSX.Element {
  const incidentals = useRecoilValue(incidentalsState);
  const selectedIncidentals = useRecoilValue(selectedInvoiceIncidentalsState);

  const handleToggle = (_incidentals: Incidentals) => () => {
    if (selectedIncidentals.find((i) => i.id === _incidentals.id)) {
      removeSelectedIncidentals(_incidentals);
    } else {
      addSelectedIncidentals(_incidentals);
    }
  };

  return (
    <List>
      {incidentals.map((_incidentals) => {
        const labelId = `checkbox-list-label-${_incidentals.id}`;

        return (
          <ListItem key={_incidentals.id} disablePadding>
            <ListItemButton
              role={undefined}
              onClick={handleToggle(_incidentals)}
              dense
            >
              <ListItemIcon>
                <Checkbox
                  edge="start"
                  checked={
                    !!selectedIncidentals.find((i) => i.id === _incidentals.id)
                  }
                  tabIndex={-1}
                  disableRipple
                  inputProps={{ 'aria-labelledby': labelId }}
                />
              </ListItemIcon>
              <ListItemText id={labelId} primary={_incidentals.name} />
            </ListItemButton>
          </ListItem>
        );
      })}
    </List>
  );
}

export default IncidentalsSelection;
