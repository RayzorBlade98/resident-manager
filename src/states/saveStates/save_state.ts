import { atom } from 'recoil';
import { IncidentalsState } from './incidentals_state';
import { InvoiceState } from './invoice_state';
import { ResidentState } from './resident_state';

export interface SaveState {
  residents: ResidentState;
  incidentals: IncidentalsState;
  invoices: InvoiceState;
}

export const saveState = atom<SaveState>({
  key: 'saveState',
  default: {
    residents: [],
    incidentals: {
      mandatoryIncidentals: [],
    },
    invoices: [],
  },
});

export default saveState;
