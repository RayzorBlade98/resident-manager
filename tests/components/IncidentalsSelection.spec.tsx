/* eslint-disable max-len */

import { RenderResult, fireEvent, render } from '@testing-library/react';
import { range } from 'lodash';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { getRecoil, setRecoil } from 'recoil-nexus';
import IncidentalsSelection from '_/components/IncidentalsSelection/IncidentalsSelection';
import { incidentalsState } from '_/states/saveStates/incidentals_state';
import invoiceGenerationState, {
  selectedInvoiceIncidentalsState,
} from '_/states/viewStates/invoice_generation_state';
import * as InvoiceGenerationStateModule from '_/states/viewStates/invoice_generation_state';
import RecoilTestWrapper from '_tests/__test_utils__/RecoillTestWrapper';
import IncidentalsBuilder from '_tests/__test_utils__/builders/incidentals_builder';

describe('IncidentalsSelection', () => {
  const addSelectedIncidentalsSpy = jest.spyOn(
    InvoiceGenerationStateModule,
    'addSelectedIncidentals',
  );
  const removeSelectedIncidentalsSpy = jest.spyOn(
    InvoiceGenerationStateModule,
    'removeSelectedIncidentals',
  );
  const incidentals = range(0, 5).map((i) => new IncidentalsBuilder()
    .withName(`Incidentals ${i}`)
    .withId(`Incidentals ${i}`)
    .build());
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <RecoilTestWrapper>
        <IncidentalsSelection />
      </RecoilTestWrapper>,
    );

    act(() => {
      setRecoil(incidentalsState, incidentals);
      setRecoil(invoiceGenerationState, (state) => ({
        ...state,
        selectedIncidentals: [],
      }));
    });
  });

  test('should handle incidentals selection correctly', () => {
    // Arrange
    const selectedIndices = [1, 2, 4];
    const expectedIncidentals = selectedIndices.map((i) => incidentals[i]);

    // Act
    const selectableIncidentals = renderResult.getAllByRole('listitem');
    selectedIndices.forEach((i) => act(() => {
      fireEvent.click(selectableIncidentals[i].firstChild!);
    }));

    // Assert
    expect(addSelectedIncidentalsSpy.mock.calls).toEqual(
      expectedIncidentals.map((i) => [i]),
    );

    const selectedIncidentals = getRecoil(selectedInvoiceIncidentalsState);
    expect(selectedIncidentals).toEqual(expectedIncidentals);
  });

  test('should handle incidentals deselection correctly', () => {
    // Arrange
    const selectedIndices = [1, 2, 4];
    const oldSelectedIncidentals = selectedIndices.map((i) => incidentals[i]);
    act(() => {
      setRecoil(invoiceGenerationState, (state) => ({
        ...state,
        selectedIncidentals: oldSelectedIncidentals,
      }));
    });

    // Act
    const selectableIncidentals = renderResult.getAllByRole('listitem');
    selectedIndices.forEach((i) => act(() => {
      fireEvent.click(selectableIncidentals[i].firstChild!);
    }));

    // Assert

    expect(removeSelectedIncidentalsSpy).toHaveBeenCalledTimes(3);
    expect(removeSelectedIncidentalsSpy.mock.calls).toEqual(
      oldSelectedIncidentals.map((i) => [i]),
    );

    const selectedIncidentals = getRecoil(selectedInvoiceIncidentalsState);
    expect(selectedIncidentals).toHaveLength(0);
  });

  test('should match snapshot', () => {
    // Assert
    expect(renderResult.container).toMatchSnapshot();
  });

  test('should match snapshot (with selected elements)', () => {
    // Arrange
    act(() => {
      setRecoil(invoiceGenerationState, (state) => ({
        ...state,
        selectedIncidentals: [1, 2, 4].map((i) => incidentals[i]),
      }));
    });

    // Assert
    expect(renderResult.container).toMatchSnapshot();
  });
});
