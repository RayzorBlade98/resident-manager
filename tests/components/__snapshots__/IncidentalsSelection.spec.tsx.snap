// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncidentalsSelection should match snapshot (with selected elements) 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding css-mu5x1o-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 0"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 0"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 0
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 1"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 1"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 2"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 2"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 3"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 3"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 3
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 4"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 4"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 4
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
  </ul>
</div>
`;

exports[`IncidentalsSelection should match snapshot 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding css-mu5x1o-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 0"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 0"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 0
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 1"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 1"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 2"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 2"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 3"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 3"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 3
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters css-178cqt3-MuiButtonBase-root-MuiListItemButton-root"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              aria-labelledby="checkbox-list-label-Incidentals 4"
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
          id="checkbox-list-label-Incidentals 4"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
          >
            Incidentals 4
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
  </ul>
</div>
`;
